WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -106,26 +106,34 @@ final class TotalsViewModel: ObservableObject, TotalsViewModelProtocol {
     var formattedOrderTotalPricePublisher: Published<String?>.Publisher { $formattedOrderTotalPrice }
     var formattedOrderTotalTaxPricePublisher: Published<String?>.Publisher { $formattedOrderTotalTaxPrice }
 
-    func calculateAmountsTapped(with cartItems: [CartItem], allItems: [POSItem]) {
-        startSyncingOrder(with: cartItems, allItems: allItems)
+    private var startPaymentOnReaderConnection: AnyCancellable?
+
+    func checkOutTapped(with cartItems: [CartItem], allItems: [POSItem]) {
+        Task { @MainActor in
+            await startSyncingOrder(with: cartItems, allItems: allItems)
+        }
     }
 
-    func startSyncingOrder(with cartItems: [CartItem], allItems: [POSItem]) {
+    private func startSyncingOrder(with cartItems: [CartItem], allItems: [POSItem]) async {
         guard CartItem.areOrderAndCartDifferent(order: order, cartItems: cartItems) else {
-            Task { @MainActor in
-                await prepareConnectedReaderForPayment()
-            }
+            await startPaymentWhenReaderConnected()
             return
         }
         // calculate totals and sync order if there was a change in the cart
-        Task { @MainActor in
-            await syncOrder(for: cartItems, allItems: allItems)
-        }
+        await syncOrder(for: cartItems, allItems: allItems)
     }
 
-    func cardPaymentTapped() {
-        Task { @MainActor in
-            await collectPayment()
+    func connectReaderTapped() {
+        Task { @MainActor [weak self] in
+            guard let self else {
+                return
+            }
+
```

Comment:
Is this needed? We're not explicitly referencing self and the compiler is fine with removing it, but if we expect the task lifespan to be longer than self, then perhaps let's add a comment so it's not removed by mistake in the future. 

```suggestion
        Task { @MainActor  in
```

Reply:
Hmm I didn't know we could do that. I wonder if it's something specific to `Task`, different to a normal closure.

We've got an implicit self here, so I thought that would only work if we strongly capture self.
It's `self.cardPresentPaymentService` – but you're absolutely right that it's not needed, so I'll make this change. 

Thanks for pointing it out.